Board resource

fitzoh opened the pull request:
TODO: Documentation

fitzoh commented on honeycombio/resource_board_test.go:
```diff
@@ -0,0 +1,131 @@
+    }
+}
+
+resource "honeycombio_board" "test" {
+    name  = "Test board from terraform-provider-honeycombio"
+    style = "list"
+     
+    query {
+        caption = "test query 0"
+        dataset = "%s"
+        query_json = data.honeycombio_query.test[0].json
+    }
+      
+    query {
+        caption = "test query 1"
+        dataset = "%s"
+        query_json = data.honeycombio_query.test[1].json
+    }
+
+}`, dataset, dataset)
```
This part look reasonable to you?

kvrhdn replied:
Yeah, I think this is great. This seems intuitive and easy to use.
I had put something similar here: https://github.com/kvrhdn/terraform-provider-honeycombio/issues/3#issue-669230480

fitzoh commented on honeycombio/resource_board.go:
```diff
@@ -0,0 +1,150 @@
+                        },
+                    },
+                },
+                Optional: true,
```
1f1ae8312a35a61c6c62b720d354fbd043ebdb2b

fitzoh commented on honeycombio/resource_board.go:
```diff
@@ -0,0 +1,150 @@
+
+    dataset := d.Get("dataset").(string)
+
+    return client.Triggers.Delete(dataset, d.Id())
```
thing I got the rest of those in 1012b9b4df528322a259de5dee2f96d202e57167

fitzoh commented on honeycombio/resource_board.go:
```diff
@@ -0,0 +1,150 @@
+    }
+
+    d.SetId(t.ID)
+    return resourceTriggerRead(d, meta)
```
ThinK i got the rest of those in 1012b9b4df528322a259de5dee2f96d202e57167

fitzoh commented on honeycombio/resource_board.go:
```diff
@@ -0,0 +1,150 @@
+                Type:     schema.TypeString,
+                Optional: true,
+                Default:  "list",
+            },
```
520695569bf4a9a9299a13be95f1e12a52fda8b5

fitzoh commented on honeycombio/resource_board.go:
```diff
@@ -0,0 +1,150 @@
+                        "query_json": {
+                            Type:     schema.TypeString,
+                            Required: true,
+                        },
```
6428744cdf507243d57f556daba10adf41ceebca

fitzoh commented:
I added a board example in 431d865a2e05ca597d0c50052626e4123507681b and discovered a fun issue with the query_spec data source.

If the filter op is `exists` or `does-not-exist`, the honeycomb API returns an error if you send any `value`, even an empty string.

Updated some code and added a test case in a387bad21ad50cb19baa67bffbda091222c325c8 to check for that.
I'm hoping the V2 SDK (https://github.com/kvrhdn/terraform-provider-honeycombio/issues/9) allows writing a validation function for that, but I wasn't seeing a way to do so now.

kvrhdn commented on honeycombio/resource_board.go:
```diff
@@ -0,0 +1,154 @@
+package honeycombio
+
+import (
+	"encoding/json"
+	"errors"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	honeycombio "github.com/kvrhdn/go-honeycombio"
```
A small formatting issue, goimports expects imports to be grouped by stdlib / other packages.
```suggestion
	"encoding/json"
	"errors"

	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
	honeycombio "github.com/kvrhdn/go-honeycombio"
```

fitzoh commented on example/board/main.tf:
```diff
@@ -0,0 +1,51 @@
+  style       = "list"
+
+
+  //Use dynamic config blocks to generate a query for each of the percentiles we're interested in
+  dynamic "query" {
```
yeah, it was new in 0.12: https://www.hashicorp.com/blog/hashicorp-terraform-0-12-preview-for-and-for-each/

fitzoh commented:
>I think, except for the imports formatting, this is ready to merge 👍🏻

👍 

>I particularly like how easy it is to share queries between triggers and boards thanks to honeycombio_query and query_json 🙂

Agreed

>We'll have to add a documentation page before releasing this, but I don't mind writing that since I did the others as well already.

Sounds good to me
